### ISSUE TITLE
[DX] Update docblock for backdrop_sort to include code examples

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#811]
The docblock for `backdrop_sort` contains broken english, and is also just really hard to follow. 

```
array $keys: An array of keys on which to be sort. Each key of the $keys array is the key by which sorting will be done. Each value of the $keys array must be one of the PHP constants sorting, either SORT_NUMERIC or SORT_STRING. If an unindexed array is passed in, each value will be assumed to be the key by which sorting will be done, and all comparisons will be done as though SORT_NUMERIC were passed in for each key.
```

We should clean up the grammar, and also add three code samples so that people understand what to pass this function. 1) simple string sort 2) simple sort on weight 3) sort on something much more complicated

**Comment by quicksketch** [comment]
Common examples that could be included:

```
@code
// Sort by a single numeric key:
backdrop_sort($my_array, array('weight' => SORT_NUMERIC));

// Short-hand sort by a numeric key:
backdrop_sort($my_array, array('weight'));

// Sort by numeric and string keys.
backdrop_sort($my_array, array('weight' => SORT_NUMERIC, 'title' => SORT_STRING));
@endcode
```

**Comment by quicksketch** [comment]
Thanks @hellomrcat! Merged into 1.x and 1.1.x.
